### PR TITLE
Add a customizable logo to QR

### LINKED ISSUE #1949: Logo in QR code everywhere (also in extensions)
Include logo in all QRs generated by the instance 
- or an extension on the instance (tpos, satspay, devices, invoices, ..)
-- or a variety ID from an extension on the instance (stall-ID, device-ID, ..)

Admins should be able to overwrite or turn off.

Default is on and pre-set to the LNbits logo
https://github.com/lnbits/lnbits/blob/3260acbcc7e907fa9c9c60d0582bf974031f917b/docs/logos/lnbits.png

### PR DESCRIPTION
**Pull request by talvasconcelos**
This PR adds the ability to have a small "logo" (png/jpg/svg) on the generated QR code on LNbits!
A new `.env` variable is introduced `LNBITS_QR_LOGO`, for an URL to the image! By default the LNbits logo is used. If using Admin UI, go to the themes tab:

admin

Here you can change the image to your own, or remove it, if you don't want to use one.

When creating an invoice on LNbits (extensions will follow after this PR is merged) the QR code will look like this:

invoice

It has little effect on the QR code readability, the image takes 15% of the QR code that has an error correction parameter that allows for 25% coverage. The default LNbits logo image is added to the static folder, at `/static/images/logos/lnbits.png`.

The change is made by wrapping the VueQRcode component, in another component `lnbits-qrcode`.

Closes #1949

### CODE REVIEW

**Comment by prusnak** [comment]
Why do you use a newly created PNG and not the SVG that is [already there](https://github.com/lnbits/lnbits/blob/3260acbcc7e907fa9c9c60d0582bf974031f917b/docs/logos/lnbits.svg)?

**Comment by talvasconcelos** [comment]
> Why do you use a newly created PNG and not the SVG that is [already there](https://github.com/lnbits/lnbits/blob/3260acbcc7e907fa9c9c60d0582bf974031f917b/docs/logos/lnbits.svg)?

I'd honestly like to use the logo/icon that is used in the favicon, for consistency! The PR is still a draft and it can use whatever file the team vouches for! Preferably in SVG format!

The file is not new, i just copied it to the static folder, to be served from there. Should have copied the SVG...

**Comment by arbadacarbaYK** [comment]
I dont lke the blur to it. Looks dirty.

**Comment by DoktorShift** [comment]
Nice work at all. Little thread would be the optic. It looks cheap, like a sticker put on top of the QR code. Maybe there is some air to tight this up.

Cheers 🙏

**Comment by arcbtc** [comment]
I think thats a shadow added (which I imagine will make qr few % less readable, can we ditch the shadow @talvasconcelos ?

**Comment by prusnak** [comment]
This PR branch is now in conflict with the `dev` branch.

**Comment by arbadacarbaYK** [comment]
If its svg now, does it have a transparent background ? Is the shadow gone ?

**Comment by talvasconcelos** [comment]
Final form of the QR with logo! 

qr

Default uses lnbits logo in svg.

File: lnbits/static/js/components.js [hunk 1]
```diff
@@ -356,6 +356,24 @@ Vue.component('lnbits-lnurlpay-success-action', {
   }
 })
 
+Vue.component('lnbits-qrcode', {
```
Comment (prusnak) [inline lnbits/static/js/components.js:359]: Do we really want to name the component `lnbits-qrcode`? 

This has a drawback that all extension would need to be updated to use the new element.

Can't we just name this component as `qrcode` and get all extension QR code with LNbits logo updated for free?

Comment (talvasconcelos) [reply lnbits/static/js/components.js:359]: they need to be changed in steps anyway, using `qrcode` as the name conflicts with the VueQRcode library, at to what I can figure out (i may be doing something wrong though)! But no all qrcodes need to change, some are wrapped in a `<q-responsive />` , some are not...

**Comment by arbadacarbaYK** [comment]
> Final form of the QR with logo!
> 
> qr
> 
> Default uses lnbits logo in svg. It uses 15% of the QR space, QR error correction (Q) allows up to 25%! It has a white square with the image inside.

bah, this is not nice

**Comment by prusnak** [comment]
Great work!

I know it's bike-shedding but I would either:
- drop the border radius of the white box
- drop the white box completely (what arbadacarbaYK suggests)

**Comment by dni** [comment]
> Great work!
> 
> I know it's bike-shedding but I would either:
> 
>     * drop the border radius of the white box
> 
>     * drop the white box completely (what arbadacarbaYK suggests)

lets do it @talvasconcelos

**Comment by arbadacarbaYK** [comment]
Pls consider the environment for the new version of the lib. 

The UI is very important for the identification of users with the product. 
We could go further than a 1 Logo for all users and extensions, see originating issue to this pr https://github.com/lnbits/lnbits/issues/1949 and also https://t.me/makerbits/14790

**Comment by prusnak** [comment]
I guess it would be nice to make `lnbits-qrcode` component to accept icon attribute, so extensions can use it too to set their own icons.

**Comment by prusnak** [comment]
I removed the white rectangle around the logo in 2baffb9627fd7ec52ca0cb63e482bcd7fe95367b so now it looks like this:

Screenshot 2023-10-09 at 16 19 54

I propose we merge this now and deal with outstanding issues (showing QR code in all of LNbits including extensions) in the latter PR.

**Comment by dni** [comment]
prettier is not recognizing wrong formatting in workflows

**Comment by dni** [comment]
now it failing! its a bug on dev! https://github.com/lnbits/lnbits/pull/2021

**Comment by prusnak** [comment]
> now it failing! its a bug on dev! #2021

rebased on top of `dev` after #2021 merge

**Review by dni — APPROVED** [review]

**Review by prusnak — APPROVED** [review]

**Review by michael1011 — APPROVED** [review]
Looks beautiful!

**Comment by prusnak** [comment]
Great work @talvasconcelos and thanks @dni for the merge! 

I've reopened https://github.com/lnbits/lnbits/issues/1949 because there is still some work to do.

**Comment by arbadacarbaYK** [comment]
> I removed the white rectangle around the logo in [2baffb9](https://github.com/lnbits/lnbits/commit/2baffb9627fd7ec52ca0cb63e482bcd7fe95367b) so now it looks like this:
> Screenshot 2023-10-09 at 16 19 54
> 
> I propose we merge the PR as is now and deal with outstanding issues (showing QR code in all of LNbits including extensions) in a future PR.

The proposed Logo in the originating PR didnt have a pink circle. With the circle its not blending into the QR.

**Comment by talvasconcelos** [comment]
I'll be honest, I don't like that the logo sits bare on top of the black squares. Looks weird and may not work for every logo.

!(startbucks)[https://www.starbucks.com.hk/media/image-20220111-032810_tcm65-76456_w1024_n.png]

!(alby)[https://getalby.com/p/whitecoastcrew.png]

I agree a icon attribute would be cool, especially in extensions!

**Comment by talvasconcelos** [comment]
Also, this just happened!!

no qr

**Comment by dni** [comment]
works for me, browser cache still tricky :D

**Comment by talvasconcelos** [comment]
> works for me, browser cache still tricky :D

shift F5 did the trick, but i don't get the default logo!

**Comment by arbadacarbaYK** [comment]
> > works for me, browser cache still tricky :D
> 
> shift F5 did the trick, but i don't get the default logo! which makes sense, since i didn't set anything, so it's opt in
> 
> path is wrong on `.env.example`, it should be `/static/images/logos/lnbits.svg`

wasnt supposed to be opt-in but defaulit is lnbits logo

**Pull request merged**